vhdbuilder/prefetch/scripts/optimize.sh

```diff
@@ -0,0 +1,284 @@
+#!/bin/bash
+set -uxo pipefail
```

Comment:
overall looks good, would like to see some retry handling built-in in steps which are retriable.

Reply:
at the moment I'd like to keep this as simple as possible - we'll get added retry logic for free once we migrate this to go internally